### PR TITLE
Re-added namequerytype and multilingual parameters to search API

### PR DESCRIPTION
**Pull request by justinjnlim**
Allowing parameters for multilingual name matching and whole vs partial name matching to query string.

### CODE REVIEW

File: tests/test_functional.py [hunk 1]
```diff
@@ -77,8 +77,13 @@
...
     ('get_html', '/langdoc/status/languages-1-3?macroarea=Eurasia&year=2018&family=', None, None),
     # Blueprint Tests
-    ('get', '/bp/api/search?bpsearch=anglai', None, '[{"glottocode": "stan1293", "iso": "eng", "name": "English", "level": "language"}, {"glottocode": "midd1317", "iso": "enm", "name": "Middle English", "level": "language"}, {"glottocode": "tsha1245", "iso": "tsj", "name": "Tshangla", "level": "language"}]'),
+    ('get', '/bp/api/search?bpsearch=anglai', None, '[{"message": "No matching languoids found for \'anglai\'"}]'),
+    ('get', '/bp/api/search?bpsearch=anglai&multilingual=true', None, '[{"glottocode": "stan1293", "iso": "eng", "name": "English", "level": "language"}, {"glottocode": "midd1317", "iso": "enm", "name": "Middle English", "level": "language"}, {"glottocode": "tsha1245", "iso": "tsj", "name": "Tshangla", "level": "language"}]'),
+    ('get', '/bp/api/search?bpsearch=klar', None, '[{"glottocode": "kumy1244", "iso": "kum", "name": "Kumyk", "level": "language"}]'),
+    ('get', '/bp/api/search?bpsearch=klar&namequerytype=whole', None, '[{"message": "No matching languoids found for \'klar\'"}]'),
```
Comment (baieric) [inline tests/test_functional.py]: Add tests for < 3 characters searched, searching a glottocode, and when there's no results. Also move all these blueprint tests to a separate test function. The function would have the same code as test_pages, just separate for organization's sake.

**Review by baieric — CHANGES_REQUESTED** [review]
just more organized unit testing needed!

File: tests/test_functional.py (hunk 1 above)
Comment (baieric) [reply tests/test_functional.py]: take a closer look at the api search function to see if there's any other test coverage we haven't thought of yet. Also we should add a comment for every one of these tests that quickly explains the scenario being tested.

Comment (justinjnlim) [reply tests/test_functional.py]: done!

**Comment by justinjnlim** [comment]
added/organized the tests!

**Review by baieric — APPROVED** [review]
fantastic, LGTM!

**Pull request merged**